src/agents_lists.erl

```diff
@@ -86,12 +68,58 @@ group_by(List) ->
                           dict:append_list(K, V, D)
                   end, dict:new(), List)).
 
--spec group_by_colour([agents()], pos_integer()) -> [agents()].
-group_by_colour(Tiles, N) ->
-    N = 2, % dividing in stripes
-    EveryNth = fun (Rest) ->
-                       [A || {I, A} <- lists:zip(lists:seq(1, length(Tiles)),
-                                                 Tiles),
-                             I rem N == Rest]
-               end,
-    lists:map(EveryNth, [I rem N || I <- lists:seq(1, N)]).
+-spec get_list(agents()) ->
+                      [agent()].
+get_list(Agents) ->
+    Agents.
+
+%% -spec get_tiles(TileWidth :: pos_integer(), environment()) ->
+%%                        [{dimension(), agents()}].
+%% get_tiles(Dist, Env) ->
+%%     W = (Env#env.world)#world.w,
+%%     Zeros = [{I, []} || I <- lists:seq(1, W, Dist)],
+%%     AssignTileToAgent = fun(A = #agent{pos={X, _, _}}) ->
+%%                                 ITile = trunc((X-1)/Dist)*Dist+1,
+%%                                 {ITile, [A]}
+%%                         end,
+%%     TiledAgents = lists:map(AssignTileToAgent, Env#env.agents),
+%%     TagTiles = group_by(TiledAgents ++ Zeros),
+%%     TagTiles.
+
+-spec get_tile(range(), environment()) ->
+  {range(), agents()}.
+get_tile(Range, Env) ->
+  {{X1, Y1, Z1}, {X2, Y2, Z2}} = Range,
+  Positions = [{X,Y,Z} ||
+    X <- lists:seq(X1,X2),
+    Y <- lists:seq(Y1,Y2),
+    Z <- lists:seq(Z1,Z2),
+    X =< (Env#env.world)#world.w, Y =< (Env#env.world)#world.h, Z =< (Env#env.world)#world.d],
```

Comment:
Line 97 is too long:     X =< (Env#env.world)#world.w, Y =< (Env#env.world)#world.h, Z =< (Env#env.world)#world.d],.